src/components/templates/MemriseChats/MemriseChats.tsx

```diff
@@ -0,0 +1,202 @@
+import UserProfilePicture from "components/molecules/UserProfilePicture";
+import firebase from "firebase/app";
+import { useSelector } from "hooks/useSelector";
+import { useUser } from "hooks/useUser";
+import React, { useState } from "react";
+import { WithId } from "utils/id";
+import { User } from "types/User";
+import { useParams } from "react-router-dom";
+import "./MemriseChats.scss";
+import UserProfileModal from "components/organisms/UserProfileModal";
+import ChatDrawer from "components/organisms/ChatDrawer";
+type Props = {
+  venueName: string;
+};
+
+const COLUMNS = 40;
+const ROWS = 25;
+
+const ROOMS = [
+  {
+    origin: {
+      row: 1,
+      column: 1,
+    },
+    width: 3,
+    height: 4,
+    image_url: "",
+  },
+  {
+    origin: {
+      row: 5,
+      column: 5,
+    },
+    width: 5,
+    height: 5,
+    image_url: "",
+  },
+  {
+    origin: {
+      row: 20,
+      column: 20,
+    },
+    width: 5,
+    height: 5,
+    image_url: "",
+  },
+];
+
+const MemriseChats = ({ venueName }: Props) => {
+  const { venueId } = useParams();
+  const { user, profile } = useUser();
+  const { venue, partygoers } = useSelector((state) => ({
+    partygoers: state.firestore.ordered.partygoers,
+    venue: state.firestore.data.currentVenue,
+  }));
+  const [selectedUserProfile, setSelectedUserProfile] = useState<
+    WithId<User>
+  >();
+
+  const partygoersBySeat: WithId<User>[][] = [];
+  partygoers.forEach((partygoer) => {
+    if (
+      !partygoer?.data ||
+      partygoer.data[venueId] === undefined ||
+      partygoer.data[venueId].row === undefined ||
+      partygoer.data[venueId].column === undefined
+    )
+      return;
+    const row = partygoer.data[venueId].row || 0;
+    const column = partygoer.data[venueId].column || 0;
+    if (!(row in partygoersBySeat)) {
+      partygoersBySeat[row] = [];
+    }
+    console.log(partygoersBySeat);
+    partygoersBySeat[row][column] = partygoer;
+  });
+
+  const takeSeat = (
+    translatedRow: number | null,
```

Comment:
nit: can be `row` and `column` since there's no translation in this template, just to remove confusion